sdk/core/azure-core-rest/src/test/java/com/azure/android/core/rest/util/paging/CFStringPageRetriever.java

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.core.rest.util.paging;
+
+import com.azure.android.core.util.Function;
+import com.azure.android.core.util.paging.Page;
+
+import org.junit.jupiter.api.Assertions;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import java9.util.concurrent.CompletableFuture;
+
+final class CFStringPageRetriever
+    implements Function<String, CompletableFuture<PagedResponse<String>>> {
```

Comment:
I need to understand why "String" is so special. Can this class be generalized to CFPageRetriever<T>?

Reply:
You mean the first param right, its the type of the nextLink (some libraries support non-string nextLink), but for most of the libraries including Chat, we will not force end-user to provide this generic type, they will work with `PagedResponse` that is specialized for `String` nextLink. Just to clarify, this file is a part of the test.